src/harness/testcases/WINNF_FT_S_PPR_testcase.py

```diff
@@ -401,3 +409,257 @@ def test_WINNF_FT_S_PPR_2(self, config_filename):
+    # Start the server
+    sas_test_harness_object.start()
+
+    # Inform SAS UUT about SAS Test Harness.
+    certificate_hash = getCertificateFingerprint(test_harness['serverCert'])
+    self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,
+                                   'url': sas_test_harness_object.getBaseUrl()})
+
+    # Store required info in the test harness.
+    self.fad_cert = test_harness['serverCert']
+    self.fad_key = test_harness['serverKey']
+    self.sas_test_harness_objects.append(sas_test_harness_object)
+
+    # Extract PPA record from peer SAS and add to local protected entities.
+    peer_sas_ppa = config['sasTestHarnessDumpRecords']['ppaRecords'][0]
+    self.protected_entity_records['ppaRecords'] = [peer_sas_ppa]
+
+    # Inject all PALs (used by SAS UUT PPA and peer SAS PPA)
+    logging.info('Step 2: inject PAL records.')
+    for index, pal_record in enumerate(config['palRecords']):
+      try:
+        logging.info('Injecting PAL record #%d', index)
+        self._sas_admin.InjectPalDatabaseRecord(pal_record)
+      except Exception:
+        logging.error(common_strings.CONFIG_ERROR_SUSPECTED)
+        raise
+    self.protected_entity_records['palRecords'] = config['palRecords']
+
+    # Register, inject PPA, and request grants.
+    logging.info('Steps 3 - 5: register, inject PPA, request grants')
+    domain_proxy_config = config['domainProxy']
+    domain_proxy = test_harness_objects.DomainProxy(self,
+                                                    domain_proxy_config['cert'],
+                                                    domain_proxy_config['key'])
+    self.domain_proxy_objects.append(domain_proxy)
+    ppa = domain_proxy.registerCbsdsAndRequestGrantsWithPpa(
+        domain_proxy_config['registrationRequests'],
+        domain_proxy_config['grantRequests'], config['ppaRecord'],
+        config['ppaClusterList'],
+        domain_proxy_config['conditionalRegistrationData'])
+    # Make sure SAS UUT's PPA is also checked for protection.
+    self.protected_entity_records['ppaRecords'].append(ppa)
+
+    # FAD exchange.
+    logging.info('Step 6 + 7: FAD exchange')
+    self.sas_uut_fad = getFullActivityDumpSasUut(self._sas, self._sas_admin,
+                                                 self.fad_cert, self.fad_key)
+    self.test_harness_fads.append(
+        getFullActivityDumpSasTestHarness(
+            self.sas_test_harness_objects[0].getSasTestHarnessInterface()))
+
+    # Trigger CPAS in SAS UUT, and wait until completion.
+    logging.info('Step 8: trigger CPAS')
+    self.cpas = self.cpas_executor.submit(
+        self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete)
+
+    logging.info('Step 9: execute IAP reference model')
+    # Pre-IAP filtering.
+    pre_iap_filtering.preIapReferenceModel(self.protected_entity_records,
+                                           self.sas_uut_fad,
+                                           self.test_harness_fads)
```

Comment:
there is one problem in the file data.py in the method getCbsdsNotPartOfPpaCluster.
as the id of CBSD records in the dump start by "cbsd/" but  the 'cbsdReferenceId' doesn't, the follwing condition will not be satisfied:
  if cbsd['id'] not in ppa_record['ppaInfo']['cbsdReferenceId']:

so during the IAP prefiltering, the reference model will not take the CBSD from SAS UUT dump as part of the PPA cluster list for the PPA of SAS UUT.

right?